src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -317,6 +335,254 @@ def on_time_step__prepare(self, event: Event):
+            "Same-sex partner",
+        ]
+
+        # Get household structure for population to vectorize choices
+        # Non-GQ population
+        gen_population = pop.loc[~pop["household_id"].isin(data_values.GQ_HOUSING_TYPE_MAP)]
+        household_structure = self.get_household_structure(gen_population)
+
+        # Children helper index groups
+        children_of_rp_idx = household_structure.loc[
+            household_structure["relation_to_household_head_x"].isin(children)
+        ].index
+        child_relative_of_rp_idx = household_structure.loc[
+            ~household_structure["relation_to_household_head_x"].isin(
+                non_relatives + children
+            )
+        ].index
+        child_non_relative_of_rp_idx = household_structure.loc[
+            household_structure["relation_to_household_head_x"].isin(non_relatives)
+        ].index
+        child_rp_idx = household_structure.loc[
+            (household_structure["relation_to_household_head_x"] == "Reference person")
+            & (household_structure["age_x"] < 18)
+        ].index
+
+        # Potential guardian/household member index groups
+        relatives_of_rp_idx = household_structure.loc[
+            ~household_structure["relation_to_household_head_y"].isin(non_relatives)
+        ].index
+        non_relatives_of_rp_idx = household_structure.loc[
+            household_structure["relation_to_household_head_y"].isin(non_relatives)
+        ].index
+        age_bound_idx = household_structure.loc[
+            (household_structure["age_difference"] >= 20)
+            & (household_structure["age_difference"] < 46)
+        ].index
+        parents_of_rp_idx = household_structure.loc[
+            household_structure["relation_to_household_head_y"].isin(parents)
+        ].index
+        over_18_idx = household_structure.loc[household_structure["age_y"] >= 18].index
+        rp_idx = household_structure.loc[
+            household_structure["relation_to_household_head_y"] == "Reference person"
+        ].index
+        partners_idx = household_structure.loc[
+            household_structure["relation_to_household_head_y"].isin(partners)
+        ].index
+
+        # Assign guardians across groups
+        # Children of reference person - red box
+        rp_parent_ids = (
+            household_structure.loc[children_of_rp_idx.intersection(rp_idx)]
+            .reset_index()
+            .set_index("child_id")["person_id"]
+        )
+        pop.loc[rp_parent_ids.index, "guardian_1"] = rp_parent_ids
+        # Assign partners of reference person as partners
+        partners_of_rp_parent_ids = household_structure.loc[
+            children_of_rp_idx.intersection(partners_idx)
+        ]
+        if len(partners_of_rp_parent_ids) > 0:
+            partners_of_rp_parent_ids = (
+                partners_of_rp_parent_ids.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
```

Comment:
it would be helpful to leave a comment here to note what you're doing re: selecting one at random if they have multiple partners